spec/javascript/packages/document-capture/hooks/use-log-camera-info-spec.tsx

```diff
@@ -0,0 +1,115 @@
+import sinon from 'sinon';
+import { render } from '@testing-library/react';
+import { useLogCameraInfo } from '@18f/identity-document-capture/hooks/use-log-camera-info';
+import { AnalyticsContextProvider } from '@18f/identity-document-capture/context/analytics';
+import { useDefineProperty } from '@18f/identity-test-helpers';
+import userEvent from '@testing-library/user-event';
+
+interface MockComponentProps {
+  isBackOfId: boolean;
+  hasStartedCropping: boolean;
+}
+
+function MockComponent({ isBackOfId, hasStartedCropping }: MockComponentProps) {
+  useLogCameraInfo({ isBackOfId, hasStartedCropping });
+  return <div>mockcomponent</div>;
+}
+
+const mockTrack = {
+  label: 'mockTrackLabel',
+  getSettings: () => ({
+    frameRate: 60,
+    height: 3000,
+    width: 3000,
+  }),
+};
+
+const mockDeviceId = 'mockDeviceId';
+const mockGetVideoTracks = () => [mockTrack];
+const mockDevice = {
+  kind: 'videoinput',
+  deviceId: mockDeviceId,
+};
+
+const mockGetUserMediaThrowsError = () => {
+  throw new Error('camera logging failed');
+};
+const mockGetUserMedia = () =>
+  new Promise((resolve) => resolve({ getVideoTracks: mockGetVideoTracks }));
+const mockEnumerateDevices = () => new Promise((resolve) => resolve([mockDevice]));
+
+describe('document-capture/hooks/use-log-camera-info', () => {
+  const defineProperty = useDefineProperty();
+  beforeEach(() => {
+    defineProperty(global.navigator, 'mediaDevices', {
+      value: {
+        getUserMedia: mockGetUserMedia,
+        enumerateDevices: mockEnumerateDevices,
+      },
+      configurable: true,
+      writable: true,
+    });
+  });
+
+  it('logs camera info when isBackOfId and hasStartedCropping are both true', async () => {
+    const trackEvent = sinon.stub();
+    const [isBackOfId, hasStartedCropping] = [true, true];
+    const { findByText } = render(
+      <AnalyticsContextProvider trackEvent={trackEvent}>
+        <MockComponent isBackOfId={isBackOfId} hasStartedCropping={hasStartedCropping} />
+      </AnalyticsContextProvider>,
+    );
+
+    // This click is not triggering anything, it's the easiest way I found to make the test wait bit until the async
+    // trackEvent can be called, I can't get 'waitFor' to work
+    await userEvent.click(await findByText('mockcomponent'));
+    expect(trackEvent).to.have.been.calledWith('idv_camera_info_logged');
+  });
+
+  it('doesnt log camera info when isBackOfId is false', async () => {
+    const trackEvent = sinon.stub();
+    const [isBackOfId, hasStartedCropping] = [false, true];
+    const { findByText } = render(
+      <AnalyticsContextProvider trackEvent={trackEvent}>
+        <MockComponent isBackOfId={isBackOfId} hasStartedCropping={hasStartedCropping} />
+      </AnalyticsContextProvider>,
+    );
+
+    // This click is not triggering anything, it's the easiest way I found to make the test wait bit until the async
+    // trackEvent can be called, I can't get 'waitFor' to work
```

Comment:
I'm not asking because this is a blocking concern, I'm asking because I want to better understand this.  Why is the click better than something like `await waitFor(() => findByText('mockcomponent'));`?   Does the click cause the test to wait for longer, and hence reduce the risk of false positives?

Reply:
Could you put the logs you got somewhere (or reproduce them?) I'm curious to see them.

Reply:
I [switched](https://github.com/18F/identity-idp/pull/10227/commits/9fb332291e6eb8a8dc80a377efd06191728b843a) to using `await waitFor(() => findByText('mockcomponent'));` which is much cleaner. Thanks!

I did try this, but it wasn't working? I may have been doing something wrong. Anyway, it works beautifully now. Thanks!

Reply:
@charleyf Yes, I just shared them with you via google drive.  Let me know if you didn't get the email notification.

Reply:
Got them, thanks. Welp, those logs are a pretty good argument for logging more information about the error 😅

Reply:
@charleyf I tested again and added the new logs to the google doc.